api/src/main/java/org/apache/iceberg/actions/SnapshotTable.java

```diff
@@ -60,6 +60,12 @@
    */
   SnapshotTable tableProperty(String key, String value);
 
+  /**
+   * @param numReaders the number of concurrent file read operations to use per partition
+   * @return this for method chaining
+   **/
+  SnapshotTable withParallelReads(int numReaders);
```

Comment:
Not a big deal, but I would consider defaulting this to unsupported so that we can review just one set of changes in the 3.2 module. I would recommend that and then doing the other modules as a follow up.